### PR TITLE
Random improvements

### PR DESCRIPTION
**Pull request by cpuguy83**
**Reason for Change**:

- Better build caching
- Custom args for e2e tests
- Detect failure on e2e/setup.sh

**Issue Fixed**:

**Notes for Reviewers**:

### CODE REVIEW

**Comment by cpuguy83** [comment]
```
E0718 18:41:07.839633   11476 mic.go:683] Applying binding testbinding node test-node for pod test-pod-default-test-id resulted in error error returned from cloud provider
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0xc27050]

goroutine 10 [running]:
github.com/Azure/aad-pod-identity/pkg/cloudprovider.(*vmIdentityInfo).GetUserIdentityList(0xc00032e040, 0x13a5740, 0xc00032e040, 0xc00032e038)
	/home/vsts/work/1/s/gopath/src/github.com/Azure/aad-pod-identity/pkg/cloudprovider/vm.go:120 +0x10
github.com/Azure/aad-pod-identity/pkg/cloudprovider.(*Client).GetUserMSIs(0xc0003b7600, 0xc0002c5b80, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/vsts/work/1/s/gopath/src/github.com/Azure/aad-pod-identity/pkg/cloudprovider/cloudprovider.go:166 +0x114
github.com/Azure/aad-pod-identity/pkg/mic.(*Client).getUserMSIListForNode(0xc0003e0af0, 0xc0002c5b80, 0x0, 0x0, 0xc0004f0380, 0x1, 0x1)
	/home/vsts/work/1/s/gopath/src/github.com/Azure/aad-pod-identity/pkg/mic/mic.go:560 +0x3e
github.com/Azure/aad-pod-identity/pkg/mic.(*Client).updateUserMSI(0xc0003e0af0, 0x0, 0x0, 0x0, 0x124bf01, 0x9, 0x0, 0x0, 0x0, 0xc0004f0360, ...)
	/home/vsts/work/1/s/gopath/src/github.com/Azure/aad-pod-identity/pkg/mic/mic.go:667 +0x1210
github.com/Azure/aad-pod-identity/pkg/mic.(*Client).updateNodeAndDeps(0xc0003e0af0, 0x0, 0x0, 0x0, 0xc0000b3b60, 0xc0004ee180)
	/home/vsts/work/1/s/gopath/src/github.com/Azure/aad-pod-identity/pkg/mic/mic.go:631 +0x14a
github.com/Azure/aad-pod-identity/pkg/mic.(*Client).Sync(0xc0003e0af0, 0xc000096c00)
	/home/vsts/work/1/s/gopath/src/github.com/Azure/aad-pod-identity/pkg/mic/mic.go:238 +0x959
github.com/Azure/aad-pod-identity/pkg/mic.(*TestMICClient).testRunSync.func1(0xc00000e688, 0xc000096c00, 0xc000096ba0)
	/home/vsts/work/1/s/gopath/src/github.com/Azure/aad-pod-identity/pkg/mic/mic_test.go:629 +0x38
created by github.com/Azure/aad-pod-identity/pkg/mic.(*TestMICClient).testRunSync
	/home/vsts/work/1/s/gopath/src/github.com/Azure/aad-pod-identity/pkg/mic/mic_test.go:628 +0xb7
FAIL	github.com/Azure/aad-pod-identity/pkg/mic	0.016s
```

File: Makefile [hunk 1]
```diff
@@ -35,7 +35,7 @@ else
 endif
 
 GO_BUILD_OPTIONS := --tags "netgo osusergo"  -ldflags "-s -X $(NMI_VERSION_VAR)=$(NMI_VERSION) -X $(MIC_VERSION_VAR)=$(MIC_VERSION) -X $(GIT_VAR)=$(GIT_HASH) -X $(BUILD_DATE_VAR)=$(BUILD_DATE) -extldflags '-static'"
-E2E_TEST_OPTIONS := -count=1 -v -timeout 24h -ginkgo.failFast 
+E2E_TEST_OPTIONS := -count=1 -v -timeout 24h -ginkgo.failFast $(E2E_TEST_OPTIONS_EXTRA)
```
Comment (kkmsft) [inline Makefile]: What is E2E_TEST_OPTIONS_EXTRA used for ?

Comment (cpuguy83) [reply Makefile]: I guess I only put it in the commit message...
e.g. `--ginko.focus=mytest`

**Comment by kkmsft** [comment]
Can you please raise a bug for the UT crash. I don't think its related to this change. Merging this change. Thank you @cpuguy83

**Pull request merged**